e2e_testing/torchscript/histogram_binning_calibration.py

```diff
@@ -0,0 +1,103 @@
+
+# Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+# Also available under a BSD-style license. See LICENSE.
+
+import torch
+
+from torch_mlir_e2e_test.torchscript.framework import TestUtils
+from torch_mlir_e2e_test.torchscript.registry import register_test_case
+from torch_mlir_e2e_test.torchscript.annotations import annotate_args, export
+
+
+# ==============================================================================
+
+class HistogramBinningCalibrationByFeature(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        # HistogramBinningCalibrationByFeature
+        self._num_segments = 42
+        self._num_bins = 5000
+        _num_interval = (self._num_segments + 1) * self._num_bins
+        _lower_bound = 0
+        _upper_bound = 1
+        l, u = _lower_bound, _upper_bound
+        w = (u - l) / self._num_bins
+        self.step = w
+        self.register_buffer("_boundaries", torch.arange(l + w, u - w / 2, w))
+        self.register_buffer(
+            "_bin_num_examples",
+            torch.empty([_num_interval], dtype=torch.float64).fill_(0.0),
+        )  # ConstantFill
+        self.register_buffer(
+            "_bin_num_positives",
+            torch.empty([_num_interval], dtype=torch.float64).fill_(0.0),
+        )  # ConstantFill
+        self.register_buffer("_bin_ids", torch.arange(_num_interval))
+        self.positive_weight = torch.tensor([0.4])
+        self.bin_ctr_in_use_after = 0
+        self.bin_ctr_weight_value = 0.9995
+        self.oneminusbin_ctr_weight_value = 0.0005
+        self._iteration = 0
+
+    @export
+    @annotate_args([
+        None,
+        ([5000], torch.int32, True),

```

Comment:
Any reason static shape is used here?

Reply:
Yes this was because with dynamic shapes an older version of the op was having trouble running correctly on IREE with CUDA backend. It was making a grid of 79x64 with static shapes but with dynamic shapes the grid was 5000x64.

Reply:
Our test cases shouldn't be influenced by the details of specific backends. So I think that making it dynamic would be better.